src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx

```diff
@@ -59,10 +71,23 @@ export const useEnvironmentSelectV2 = (params: UseEnvironmentSelectV2Params) =>
           return
         }
       }
+
       if (!noDefault) {
-        setSelectedEnvironment(selectOptions[0])
-        onChange(selectOptions[0], data?.data?.content?.[0], false)
-        rewriteCurrentLocationWithActiveEnvironment(data?.data?.content?.[0].identifier)
+        const preferenceOption = selectOptions.find(({ value }) => value === preferredEnvironment)
+
+        if (preferenceOption) {
+          setSelectedEnvironment(preferenceOption)
+          onChange(
+            preferenceOption,
+            data.data.content.find(({ identifier }) => identifier === preferredEnvironment) as EnvironmentResponseDTO,
```

Comment:
```suggestion
            data.data.content?.find(({ identifier }) => identifier === preferredEnvironment) as EnvironmentResponseDTO,
```
do we need to do this ^ here?

Reply:
Cool thanks for explaining

Reply:
I actually did have it with a bunch of `?`, but then took a second look, realised they weren't needed and TS didn't disagree :-)